support/BRFileService.c

```diff
@@ -260,213 +599,181 @@ fileServiceLoad (BRFileService fs,
+            return fileServiceFailedImpl (fs, 1, (dataBytes == dataBytesBuffer ? NULL : dataBytes), NULL,
+                                          "missed bytes count");
+        }
+
+        entityBytes = &dataBytes[offset];
+
+        switch (headerVersion) {
+            case HEADER_FORMAT_1:
+                // compute then compare checksum
+                break;
+        }
+
+        // Look up the entity handler
+        BRFileServiceEntityHandler *handler = fileServiceEntityTypeLookupHandler(entityType, version);
+        if (NULL == handler)
+            return fileServiceFailedImpl (fs, 1, (dataBytes == dataBytesBuffer ? NULL : dataBytes), NULL,
+                                          "missed type handler");
+
+        // Read the entity from buffer and add to results.
+        void *entity = handler->reader (handler->context, fs, entityBytes, entityBytesCount);
+        if (NULL == entity)
+            return fileServiceFailedEntity (fs, 1, (dataBytes == dataBytesBuffer ? NULL : dataBytes), NULL,
+                                            type, "reader");
+
+        // Update restuls with the newly restored entity
+        BRSetAdd (results, entity);
+
+        // If the read version is not the current version, update
+        if (updateVersion &&
+            (version != entityType->currentVersion ||
+             headerVersion != currentHeaderFormatVersion))
+            _fileServiceSave (fs, type, entity, 0);
     }
+    sqlite3_reset (fs->sdbSelectAllStmt);
+    pthread_mutex_unlock (&fs->lock);
 
-    if (0 != fclose (file)) {  fileServiceFailedUnix (fs, bytes, NULL, errno); return; }
+    if (dataBytes != dataBytesBuffer) free (dataBytes);
 
-    free (bytes);
+    return 1;
 }
 
 /// MARK: - Remove, Clear
 
-extern void
+extern int
 fileServiceRemove (BRFileService fs,
                    const char *type,
                    UInt256 identifier) {
     BRFileServiceEntityType *entityType = fileServiceLookupType (fs, type);
-    if (NULL == entityType) { fileServiceFailedImpl (fs, NULL, NULL, "missed type"); return; };
+    if (NULL == entityType)
+        return fileServiceFailedImpl (fs, 0, NULL, NULL, "missed type");
+
+    // Hex-Encode identifier
+    const char *hash = u256hex(identifier);
+
+    sqlite3_status_code status;
+
+    pthread_mutex_lock (&fs->lock);
+
+    status = sqlite3_bind_text (fs->sdbDeleteStmt, 1, type, -1, SQLITE_STATIC);
+    if (SQLITE_OK != status)
+        return fileServiceFailedSDB (fs, 1, status);
+
+    status = sqlite3_bind_text (fs->sdbDeleteStmt, 2, hash, -1, SQLITE_STATIC);
+    if (SQLITE_OK != status)
+        return fileServiceFailedSDB (fs, 1, status);
+
+    if (SQLITE_DONE != sqlite3_step (fs->sdbDeleteStmt))
+        return fileServiceFailedSDB (fs, 1, status);
 
-    char filename[strlen(fs->pathToType) + 1 + strlen(type) + 1 + 2*sizeof(UInt256) + 1];
-    sprintf (filename, "%s/%s/%s", fs->pathToType, type, u256hex(identifier));
+    sqlite3_reset (fs->sdbDeleteStmt);
```

Comment:
Same comment re: is this sqlite3_reset needed in the early return cases?

Reply:
I'll change these to be 'on entry' - ensure we start cleanly on each function entry.